test/commands/prodly/nut/checkIn.nut.js

```diff
@@ -0,0 +1,99 @@
+import { execCmd } from '@salesforce/cli-plugins-testkit';
+import { expect } from 'chai';
+describe('CLI CheckIn Commands Automation', () => {
+  // 1. Check-in With all the correct Flag
+  it('Check-In commands with all the flags using Dataset flag with Name', () => {
+    const rv = execCmd(
+      "prodly:checkin -i 'c31b1cde-7520-4f93-8b54-7347415764f3' -b 'main' -t 'Automation Dataset Account' -c 'Automation Test CLI checkin with all the flags' --json"
+    );
+    const response = JSON.parse(rv.shellOutput.stdout);
+    expect(response.result.jobId).to.exist;
+    expect(response.result.message).to.be.string('Checkin launched');
+  });
+
+  // 2. Check-in Without Instance flag 
+  it('Check-In commands Without Instance flag using Dataset flag with Name', () => {
+    const rv = execCmd(
+      "prodly:checkin -b 'main' -t 'Automation Dataset Account' -c 'Automation Test CLI checkin with Without Instance flag' --json"
+    );
+    const response = JSON.parse(rv.shellOutput.stdout);
+    expect(response.result.jobId).to.exist;
+    expect(response.result.message).to.be.string('Checkin launched');
+  });
+
+  //3. Check-in With all the correct Flag
+  it('Check-In commands without branch flag using Dataset flag with Name', () => {
+    const rv = execCmd(
+      "prodly:checkin -i 'c31b1cde-7520-4f93-8b54-7347415764f3' -t 'Automation Dataset Account' -c 'Automation Test CLI' --json"
+    );
+    const response = JSON.parse(rv.shellOutput.stdout);
+    expect(response.result.jobId).to.exist;
+    expect(response.result.message).to.be.string('Checkin launched');
+  });
+  // 4. Check-in Without dataset flag
+  it('Check-In commands Without dataset flag with Name flag', () => {
+    const rv = execCmd(
+      "prodly:checkin -i 'c31b1cde-7520-4f93-8b54-7347415764f3' -b 'main' -c 'Automation Test CLI' --json"
```

Comment:
Move the dataset name into an env variable. We will then setup the variable in github action too.